sam.c

```diff
@@ -1911,6 +1911,12 @@ static sam_hdr_t *sam_hdr_sanitise(sam_hdr_t *h) {
         cp[h->l_text] = '\0';
     }
 
+     if (sam_hdr_fill_hrecs(h) < 0) {
```

Comment:
I'm not sure if we want to call `sam_hdr_fill_hrecs()` here.  It will trigger a full parse of the header, which we may not want to do unless really needed.

Reply:
removed